synapse/handlers/register.py

```diff
@@ -149,7 +162,7 @@ def register(
                     user_id = None
                     token = None
                     attempts += 1
-                    if attempts > 5:
+                    if attempts > 50:
```

Comment:
Tidied up a bit. Reduced the scope of what can SynapseError. Removed the limit.

I could push the logic into a few different places (e.g. have the ID allocation be done in the store), but I don't care enough to make more invasive changes.

4a688a3